django/db/models/manager.py

```diff
@@ -242,12 +244,40 @@ class Manager(BaseManager.from_queryset(QuerySet)):
 class ManagerDescriptor(object):
     # This class ensures managers aren't accessible via model instances.
     # For example, Poll.objects works, but poll_obj.objects raises AttributeError.
-    def __init__(self, manager):
+    # It also ensures managers defined on non-abstract base classes are not inherited by child classes.
+    def __init__(self, manager, model):
         self.manager = manager
+        self.model = model
 
     def __get__(self, instance, cls=None):
         if instance is not None:
             raise AttributeError("Manager isn't accessible via %s instances" % cls.__name__)
+
+        if cls._meta.concrete_model == self.model._meta.concrete_model:
```

Comment:
I don't think you need this check here. This case will be handled in the for loop. Otherwise you can skip the first item in `cls.mro()` as well.